google-cloud-spanner/src/test/java/com/google/cloud/spanner/SessionPoolStressTest.java

```diff
@@ -124,55 +127,67 @@ private void setupSpanner(DatabaseId db) {
             Mockito.anyInt(), Mockito.anyBoolean(), Mockito.any(SessionConsumer.class));
   }
 
-  private void setupSession(final SessionImpl session) {
-    ReadContext mockContext = mock(ReadContext.class);
+  SessionImpl getMockedSession(ReadContext context) {
+    SpannerImpl spanner = mock(SpannerImpl.class);
+    Map options = new HashMap<>();
+    options.put(Option.CHANNEL_HINT, channelHint.getAndIncrement());
+    final SessionImpl session =
+        new SessionImpl(
+            spanner,
+            "projects/dummy/instances/dummy/databases/dummy/sessions/session" + sessionIndex,
+            options) {
+          @Override
+          public ReadContext singleUse(TimestampBound bound) {
+            // The below stubs are added so that we can mock keep-alive.
+            return context;
+          }
+
+          @Override
+          public ApiFuture<Empty> asyncClose() {
+            synchronized (lock) {
+              if (expiredSessions.contains(this.getName())) {
+                return ApiFutures.immediateFailedFuture(
+                    SpannerExceptionFactoryTest.newSessionNotFoundException(this.getName()));
+              }
+              if (sessions.remove(this.getName()) == null) {
+                setFailed(closedSessions.get(this.getName()));
+              }
+              closedSessions.put(this.getName(), new Exception("Session closed at:"));
+              if (sessions.size() < minSessionsWhenSessionClosed) {
+                minSessionsWhenSessionClosed = sessions.size();
+              }
+            }
+            return ApiFutures.immediateFuture(Empty.getDefaultInstance());
+          }
+
+          @Override
+          public void prepareReadWriteTransaction() {
```

Comment:
Probably for a different PR, but this seems like a left-over from before InlineBeginTransaction.